### PR TITLE
#9607: Endpoint path must support URI path as per rfc3986 appendix-B

### LINKED ISSUE #9607: Allow the use of dashes in management endpoint names
### Feature description

Consider below management endpoint definition:
```
import io.micronaut.management.endpoint.annotation.Endpoint;

@Endpoint("service-metadata")
public class ServiceMetadataController {
... 
}
```

### Current behaviour:
When defining a new management endpoint that contains dashes (e.g. /service-metadata) does not work, since when accessing this endpoint the application returns 404.

### Desired behaviour:
Allow defining management endpoints containing dashes. 
Ideally it should allow using any name for the management endpoint that is a valid URL path. 

The management endpoint id is verified against [this regex](https://github.com/micronaut-projects/micronaut-core/blob/4.0.x/management/src/main/java/io/micronaut/management/endpoint/processors/AbstractEndpointRouteBuilder.java#L52).

### PR DESCRIPTION
**Pull request by pragyamutluru**
Fixes https://github.com/micronaut-projects/micronaut-core/issues/9607
I have referred to DefaultUriBuilder.STRING_PATTERN_PATH to change the regex being checked in AbstractEndpointRouteBuilder.ENDPOINT_ID_PATTERN.

### CODE REVIEW

**Comment by sdelamo** [comment]
@pragyamutluru you need to sign the contributor license agreement

File: management/src/main/java/io/micronaut/management/endpoint/processors/AbstractEndpointRouteBuilder.java [hunk 1]
```diff
@@ -49,7 +49,7 @@
 @Internal
 abstract class AbstractEndpointRouteBuilder extends DefaultRouteBuilder implements ExecutableMethodProcessor<Endpoint>, LifeCycle<AbstractEndpointRouteBuilder> {
 
-    private static final Pattern ENDPOINT_ID_PATTERN = Pattern.compile("\\w+");
+    private static final Pattern ENDPOINT_ID_PATTERN = Pattern.compile("([^#?]*)");
```
Comment (yawkat) [inline management/src/main/java/io/micronaut/management/endpoint/processors/AbstractEndpointRouteBuilder.java]: I think this is too wide. 
```suggestion
    private static final Pattern ENDPOINT_ID_PATTERN = Pattern.compile("[\\w-]+");
```

**Review by yawkat — CHANGES_REQUESTED** [review]

File: management/src/main/java/io/micronaut/management/endpoint/processors/AbstractEndpointRouteBuilder.java (hunk 1 above)
Comment (graemerocher) [reply management/src/main/java/io/micronaut/management/endpoint/processors/AbstractEndpointRouteBuilder.java]: Agreed, what pattern do you suggest?

Comment (pragyamutluru) [reply management/src/main/java/io/micronaut/management/endpoint/processors/AbstractEndpointRouteBuilder.java]: Updated code to narrow down the check as per suggestion. Kindly check the latest change.

**Review by yawkat — APPROVED** [review]

**Review by graemerocher — APPROVED** [review]

**Comment by graemerocher** [comment]
Thanks for the contribution!

**Pull request merged**